netty/src/test/java/io/grpc/netty/TlsTest.java

```diff
@@ -185,6 +186,54 @@ public void basicClientServerIntegrationTest() throws Exception {
     client.unaryRpc(SimpleRequest.getDefaultInstance());
   }
 
+  /**
+   * Test that a client fails to send requests over TLS to a server with an unexpected hostname
+   * listed in an otherwise valid certificate.
+   */
+  @Test
+  public void serverRejectedBadHostname() throws Exception {
```

Comment:
Client does the rejection. Can we rename to `clientRejectedMismatchedHostname` ?